pkg/registry/resourcequota/registry.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcequota
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/resourcequotausage"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+)
+
+// Registry implements operations to modify ResourceQuota objects
+type Registry interface {
+	generic.Registry
+	resourcequotausage.Registry
+}
+
+// registry implements custom changes to generic.Etcd.
+type registry struct {
+	*etcdgeneric.Etcd
+}
+
+// ApplyStatus atomically updates the ResourceQuotaStatus based on the observed ResourceQuotaUsage
```

Comment:
It is modifying one key, but the modification is mandating an etcd atomic compare and swap based on the resource version. 

Sent from my iPhone

> On Jan 27, 2015, at 3:50 AM, Eric Tune <notifications@github.com> wrote:
> 
> In pkg/registry/resourcequota/registry.go:
> 
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
> > +)
> > +
> > +// Registry implements operations to modify ResourceQuota objects
> > +type Registry interface {
> > +	generic.Registry
> > +	resourcequotausage.Registry
> > +}
> > +
> > +// registry implements custom changes to generic.Etcd.
> > +type registry struct {
> > +	*etcdgeneric.Etcd
> > +}
> > +
> > +// ApplyStatus atomically updates the ResourceQuotaStatus based on the observed ResourceQuotaUsage
> What is noteworthily "atomic" about this? There is only one etcd key operated on, right, so of course it is atomic?
> 
> —
> Reply to this email directly or view it on GitHub.
>